libs/vkd3d/state.c

```diff
@@ -4364,7 +4364,37 @@ static HRESULT d3d12_pipeline_state_graphics_create_shader_stages(
     return S_OK;
 }
 
-static void d3d12_pipeline_state_graphics_handle_meta(struct d3d12_pipeline_state *state,
+static bool d3d12_pipeline_state_validate_gs_input_toplogy(struct d3d12_pipeline_state *state,
+        const struct vkd3d_shader_meta *gs_meta, uint32_t geometry_meta)
+{
+    struct d3d12_graphics_pipeline_state *graphics = &state->graphics;
+
+    switch (gs_meta->gs_input_topology)
+    {
+        case VK_PRIMITIVE_TOPOLOGY_POINT_LIST:
```

Comment:
Is this correct for tessellation? I think geometry_meta needs to reflect the meta of the TES here, not PSO input.

Reply:
Ah. No, geometry_meta is updated on line 4453.

Reply:
we're checking against TES in topologies that can actually be generated by TES (`geometry_meta` will implicitly reflect that). There shouldn't be any way for TES to emit points.